plugin/src/main/groovy/com/novoda/staticanalysis/internal/detekt/DetektConfigurator.groovy

```diff
@@ -45,18 +48,31 @@ class DetektConfigurator implements Configurator {
             }
 
             def detekt = project.extensions.findByName('detekt')
+            setDefaultXmlReport(detekt)
             config.delegate = detekt
             config()
 
             def collectViolations = configureToolTask(detekt)
             evaluateViolations.dependsOn collectViolations
         }
     }
-    
-    private CollectDetektViolationsTask configureToolTask(detekt) {
+
+    private void setDefaultXmlReport(detekt) {
+        if (detekt.hasProperty('reports')) {
+            detekt.reports {
+                xml.enabled = true
+                xml.destination = new File(project.buildDir, 'reports/detekt/detekt.xml')
```

Comment:
So this means that we always set it by default. Detekt's lazy fallback will never be used. 

But that's fine because in case users but their `destionation`, that will be used anyways.